cuckoo/web/static/js/cuckoo/rdp.js

```diff
@@ -1,2 +1,817 @@
+                }, u, u.exports, e, t, n, o)
+            }
+            return n[s].exports
+        }
+
+        for (var r = "function" == typeof require && require, s = 0; s < o.length; s++) i(o[s]);
+        return i
+    }
+
+    return e
+}()({
+    1: [function (e, t, n) {
+        "use strict";
+
+        function o(e) {
+            return e && e.__esModule ? e : {default: e}
+        }
+
+        function i(e, t) {
+            if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function")
+        }
+
+        function r(e, t) {
+            if (!e) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
+            return !t || "object" != typeof t && "function" != typeof t ? e : t
+        }
+
+        function s(e, t) {
+            if ("function" != typeof t && null !== t) throw new TypeError("Super expression must either be null or a function, not " + typeof t);
+            e.prototype = Object.create(t && t.prototype, {
+                constructor: {
+                    value: e,
+                    enumerable: !1,
+                    writable: !0,
+                    configurable: !0
+                }
+            }), t && (Object.setPrototypeOf ? Object.setPrototypeOf(e, t) : e.__proto__ = t)
+        }
+
+        Object.defineProperty(n, "__esModule", {value: !0});
+        var a = function () {
+            function e(e, t) {
+                for (var n = 0; n < t.length; n++) {
+                    var o = t[n];
+                    o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, o.key, o)
+                }
+            }
+
+            return function (t, n, o) {
+                return n && e(t.prototype, n), o && e(t, o), t
+            }
+        }(), c = e("./Hookable"), l = o(c), u = function (e) {
+            function t(e) {
+                i(this, t);
+                var n = r(this, (t.__proto__ || Object.getPrototypeOf(t)).call(this));
+                if (n.hooks = {connect: [], error: [], end: []}, !window.Guacamole) {
+                    var o;
+                    return console.error("No Guacamole! Did you forget to process the avocados in src/scripts/rdp/guac?"), o = !1, r(n, o)
+                }
+                return n.display = e.display, n.parent = e.client, n.client = null, n._mouse = null, n._keyboard = null, n
+            }
+
+            return s(t, e), a(t, [{
+                key: "connect", value: function () {
+                    let token = Cookies.get("csrftoken");
+                    if (!token) {
+                        let field = $("input[name=csrfmiddlewaretoken]");
+                        if (field && field.val()) {
+                            token = field.val();
+                        }
+                    }
+                    if (token) {
+                        var t = new Guacamole.HTTPTunnel("tunnel/", false, {"X-CSRFToken": token});
```

Comment:
In `guac.js`, adding extra headers to HTTPTunnel isn't possible. See [this line](https://github.com/cuckoosandbox/cuckoo/blob/5c775df8a9890aebfcf9f415a0d8275157ee1972/cuckoo/web/static/js/guac.js#L11527). I think you based your PR on [latest `guacamole-common-js` documentation](https://guacamole.apache.org/doc/guacamole-common-js/Guacamole.HTTPTunnel.html). To use this functionnality, we need to update `guac.js` to latest version I think.